File: src/main/java/ru/igrey/dev/domain/Deal.java
package ru.igrey.dev.domain;

import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import ru.igrey.dev.entity.RisasRequisitionEntity;

import java.util.List;

@RequiredArgsConstructor
@Getter
@EqualsAndHashCode
public class Deal {
    private final Long id;
    private final Long dealStatusId;
    private final Cas mik;
    private final Long officeId;
    private final Long typeId;
    private final Boolean archived;
    private final Cas author;
    private final String crmTasks;
    private final Documents documents;
    private final List<ServiceDeal> services;
    private final List<String> schedules;
    private final List<RisasRequisitionEntity> risasRequisitions;

    @Override
    public String toString() {
        return header()
                + "Автор (" + (author == null ? "ERROR" : author) + ")\n"
                + "МИК (" + (mik == null ? "ERROR" : mik) + ", officeId:" + officeId + ")\n"
                + "Запись: " + (schedules.isEmpty() ? "EMPTY" : schedules) + "\n"
                + "CRM: " + crmTasks + "\n"
                + "ПЭ: " + risasRequisitions + "\n"
                + "Сервисы: " + services + "\n"
                + documents.toString();
    }

    private String header() {
        String title = "Неизвестный тип сделки";
        String archivedStr = archived ? ", archived" : "";
        if (typeId == 55561) title = "Упрощеная сделка";
        if (typeId == 55562) title = "СПК";
        if (typeId == 55560) title = "Неипотечная сделка";
        if (typeId == 55550) title = "Ипотечная сделка";
        return title + " (№ " + id + ", status: " + dealStatusId + archivedStr + ")" + "\n";
    }
}

File: src/main/kotlin/ru/igrey/dev/domain/Deal.kt
package ru.igrey.dev.domain

import com.google.gson.annotations.SerializedName
import org.apache.commons.lang3.StringUtils
import ru.igrey.dev.entity.DealHistoryEntity
import ru.igrey.dev.entity.DocumentEntity
import ru.igrey.dev.entity.RisasRequisitionEntity
import java.time.LocalDateTime

data class Deal(
    val id: Long,
    val dealStatusId: Long,
    val mik: Cas? = null,
    val officeId: Long? = null,
    val typeId: Long,
    val archived: Boolean,
    val author: Cas?,
    val crmTasks: String? = null,
    val documents: Documents? = null,
    val services: List<ServiceDeal> = emptyList(),
    val schedules: List<String> = emptyList(),
    val risasRequisitions: List<RisasRequisitionEntity> = emptyList(),
    val lastHistory: List<DealHistoryEntity> = emptyList(),
    val createdTime: LocalDateTime
) {
    override fun toString(): String {
        return (header()
                + "Автор (" + (author ?: "ERROR") + ")\n"
                + "МИК (" + (mik ?: "ERROR") + ", officeId:" + officeId + ")\n"
                + "Запись: " + (if (schedules.isEmpty()) "EMPTY" else schedules) + "\n"
                + "CRM: " + crmTasks + "\n"
                + "ПЭ: " + risasRequisitions + "\n"
                + "Сервисы: " + services + "\n"
                + "История: " + lastHistory + "\n"
                + documents!!.toString())
    }

    private fun header(): String {
        val archivedStr = if (archived) ", archived" else ""
        val title = when (typeId) {
            55561L -> "Упрощеная сделка"
            55562L -> "СПК"
            55560L -> "Неипотечная сделка"
            55550L -> "Ипотечная сделка"
            else -> typeId
        }
        return "$title (№ $id, status: $dealStatusId$archivedStr, $createdTime)\n"
    }
}

data class ServiceDeal(
    @SerializedName(value = "service_type_id")
    private val id: Long,
    private val active: Boolean? = null
) {
    override fun toString() = id.toString() + ""
}

data class Documents(
    val notAuthorDocuments: List<DocumentEntity> = emptyList(),
    val uploadedDocsCount: Int
) {
    override fun toString(): String {
        val title = ("\nКоличество загруженных доков: " + uploadedDocsCount + "\n"
                + "Сгенерированнные:\n")

        val documnentsAsSb = StringBuilder()
        for (doc in notAuthorDocuments) {
            if (doc.archived && doc.documentTypeId == 20580L) continue
            documnentsAsSb.append(doc.toString())
                .append("\n" + StringUtils.repeat("-", 30) + "\n")
        }
        return title + documnentsAsSb.toString()
    }
}

